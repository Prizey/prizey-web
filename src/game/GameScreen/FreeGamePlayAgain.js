import React from 'react'
import { connect } from 'react-redux'
import { Button, Typography } from '@material-ui/core'

import UserBalance from 'design/UserBalance'
import Layout from 'design/Layout/Layout'
import ProductImage from 'design/ProductImage'
import Caption from 'design/Caption/Caption'
import placeholder from './cardi_placeholder.png'

const Screen = ({ currentUser, location, product }) => {
  const mainText = product
    ? product.title
    : `Looks like you've already played this game, Click the "Play Again" below to Download Another App and play a new game`
  const productToShow = product || {
    image: placeholder,
  }

  return (
    <Layout
      location={location}
      currentUser={currentUser}
      leftIcon={<UserBalance />}
      confirmLeave
      caption={<Caption difficulty="easy" />}
    >
      <Typography align="center" variant="h5">
        {mainText}
      </Typography>

      <ProductImage product={productToShow} />

      <a
        href="https://www.verifyspot.net/cl.php?id=c6ab41e68cc1dd56233aa14ec2ed9859"
        target="_blank"
        rel="noopener noreferrer"
        style={{ textDecoration: 'none' }}
      >
        <Button
          variant="contained"
          style={{ backgroundColor: '#76fc03' }}
          fullWidth
        >
          Play Again
        </Button>
      </a>
    </Layout>
  )
}

export const mapState = ({ basket = {} }) => ({
  product: basket.product,
})

export default connect(mapState)(Screen)
