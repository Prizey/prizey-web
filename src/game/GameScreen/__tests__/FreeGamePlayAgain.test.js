import React from 'react'
import renderer from 'react-test-renderer'

import FreeGamePlayAgain, { mapState } from '../FreeGamePlayAgain'

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component {...props} state={{ foo: 'bar' }} />
  ),
}))

jest.mock('design/UserBalance', () => props => (
  <div {...props}>UserBalance - {props.children}</div>
))

jest.mock('design/Layout/Layout', () => props => (
  <div {...props}>Layout - {props.children}</div>
))

jest.mock('design/ProductImage', () => props => (
  <div {...props}>ProductImage - {props.children}</div>
))

jest.mock('design/Caption/Caption', () => props => (
  <div {...props}>Caption - {props.children}</div>
))

describe('When there is a product', () => {
  it('renders correctly', () => {
    const props = {
      currentUser: { id: 1, name: 'foo' },
      location: 'foo location',
      product: { image: 'https://foo.image', title: 'foo title' },
    }
    const tree = renderer.create(<FreeGamePlayAgain {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('When there is no product', () => {
  it('renders correctly', () => {
    const props = {
      currentUser: { id: 1, name: 'foo' },
      location: 'foo location',
    }
    const tree = renderer.create(<FreeGamePlayAgain {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('mapState', () => {
  it('works correctly', () => {
    const product = mapState({ basket: { product: 'foo' } })

    expect(product).toMatchObject({ product: 'foo' })
  })
})
