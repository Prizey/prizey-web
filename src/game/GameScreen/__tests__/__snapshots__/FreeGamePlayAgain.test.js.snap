// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When there is a product renders correctly 1`] = `
<div
  caption={
    <Unknown
      difficulty="easy"
    />
  }
  confirmLeave={true}
  currentUser={
    Object {
      "id": 1,
      "name": "foo",
    }
  }
  leftIcon={<Unknown />}
  location="foo location"
>
  Layout - 
  <h5
    className="MuiTypography-root-1 MuiTypography-h5-17 MuiTypography-alignCenter-24"
  >
    foo title
  </h5>
  <div
    product={
      Object {
        "image": "https://foo.image",
        "title": "foo title",
      }
    }
  >
    ProductImage - 
  </div>
  <a
    href="https://www.verifyspot.net/cl.php?id=c6ab41e68cc1dd56233aa14ec2ed9859"
    rel="noopener noreferrer"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <button
      className="MuiButtonBase-root-63 MuiButton-root-37 MuiButton-contained-48 MuiButton-raised-51 MuiButton-fullWidth-62"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#76fc03",
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-38"
      >
        Play Again
      </span>
      <span
        className="MuiTouchRipple-root-66"
      />
    </button>
  </a>
</div>
`;

exports[`When there is no product renders correctly 1`] = `
<div
  caption={
    <Unknown
      difficulty="easy"
    />
  }
  confirmLeave={true}
  currentUser={
    Object {
      "id": 1,
      "name": "foo",
    }
  }
  leftIcon={<Unknown />}
  location="foo location"
>
  Layout - 
  <h5
    className="MuiTypography-root-1 MuiTypography-h5-17 MuiTypography-alignCenter-24"
  >
    Looks like you've already played this game, Click the "Play Again" below to Download Another App and play a new game
  </h5>
  <div
    product={
      Object {
        "image": "cardi_placeholder.png",
      }
    }
  >
    ProductImage - 
  </div>
  <a
    href="https://www.verifyspot.net/cl.php?id=c6ab41e68cc1dd56233aa14ec2ed9859"
    rel="noopener noreferrer"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    target="_blank"
  >
    <button
      className="MuiButtonBase-root-63 MuiButton-root-37 MuiButton-contained-48 MuiButton-raised-51 MuiButton-fullWidth-62"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#76fc03",
        }
      }
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-38"
      >
        Play Again
      </span>
      <span
        className="MuiTouchRipple-root-66"
      />
    </button>
  </a>
</div>
`;
